pipeline.go

```diff
@@ -9,6 +9,17 @@ import (
 
 type pipelineExecer func([]Cmder) error
 
+type Pipelineable interface {
```

Comment:
Can you also add runtime type assertion under `Pipeline` definition: `var _ Pipeliner = (*Pipeline)(nil)`